lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -492,11 +491,11 @@ performSelection minCoinValueFor costFor criteria
     -- ada-only inputs are available.
     --
     makeChangeRepeatedly
-        :: [Set AssetId]
-        -> SelectionState
+        :: SelectionState
         -> m (Either SelectionError (SelectionResult TokenBundle))
-    makeChangeRepeatedly changeSkeleton s@SelectionState{selected,leftover} = do
+    makeChangeRepeatedly s@SelectionState{selected,leftover} = do
         let inputsSelected = mkInputsSelected selected
+        let changeSkeleton = NE.toList $ predictChange selected
```

Comment:
I've moved this here inside the function body because now, it is possible for the predicted change to change on each iteration due to the selection of new multi-asset outputs.